src/PartitionGraph.f90

```diff
@@ -2209,4 +2229,187 @@ SUBROUTINE swapDomain(thisGraph,iv,linL1,D,lswap)
+      ENDDO !ivt
+      !Move smallest chunks to L2 until L1 is contiguous
+      DO WHILE(nchunk > 1)
+        istt=1; istp=nv1
+        !Find current smallest chunk (by weight)
+        schunkWeight=MINVAL(chunkWeight)
+        DO schunk=1,ndc
+          IF(chunkWeight(schunk) == schunkWeight) THEN
+            nv=0
+            DO ivt=istt,istp
+              iv=Order(ivt)
+              IF(ichunk(iv) == schunk) nv=nv+1
+            ENDDO !ivt
+            EXIT
+          ENDIF
+        ENDDO !schunk
+
+        !Move all vertices in this chunk to end of the list
+        OrderCpy=Order
+        iv1=0; iv2=0
+        DO ivt=istt,istp
+          iv=OrderCpy(ivt)
+          IF(ichunk(iv) /= schunk) THEN
+            iv1=iv1+1
+            Order(iv1)=iv
+          ELSE
+            iv2=iv2+1
+            Order(nv1-nv+iv2)=iv
+          ENDIF
+        ENDDO !ivt
+        chunkWeight(schunk)=HUGE(1.0_SRK)
+        nchunk=nchunk-1
+        nv1=nv1-nv
+      ENDDO !nchunk > 1
+      DEALLOCATE(chunkWeight)
+
+      !Contiguity check on L2
+      nv2=thisGraph%nvert-nv1
+      nv=nv2; istt=nv1+1; istp=thisGraph%nvert
+      CALL checkContiguity(thisGraph, nv, Order(istt:istp), ndc, ichunk)
+      nchunk=ndc
+      ALLOCATE(chunkWeight(nchunk))
+      chunkWeight=0.0_SRK
+      DO ivt=istt,istp
+        iv=Order(ivt)
+        chunkWeight(ichunk(iv))=chunkWeight(ichunk(iv))+thisGraph%wts(iv)
+      ENDDO !ivt
+      !Move smallest chunks to L2 until L1 is contiguous
+      DO WHILE(nchunk > 1)
+        istt=nv1+1; istp=thisGraph%nvert
+        !Find current smallest chunk (by weight)
+        schunkWeight=MINVAL(chunkWeight)
+        DO schunk=1,ndc
+          IF(chunkWeight(schunk) == schunkWeight) THEN
+            nv=0
+            DO ivt=istt,istp
+              iv=Order(ivt)
+              IF(ichunk(iv) == schunk) nv=nv+1
+            ENDDO !ivt
+            EXIT
+          ENDIF
+        ENDDO !schunk
+
+        !Move all vertices in this chunk to start of the list
+        OrderCpy=Order
+        iv1=0; iv2=0
+        DO ivt=istt,istp
+          iv=OrderCpy(ivt)
+          IF(ichunk(iv) /= schunk) THEN
+            iv1=iv1+1
+            Order(nv1+nv+iv1)=iv
+          ELSE
+            iv2=iv2+1
+            Order(nv1+iv2)=iv
+          ENDIF
+        ENDDO !ivt
+        chunkWeight(schunk)=HUGE(1.0_SRK)
+        nchunk=nchunk-1
+        nv1=nv1+nv
+      ENDDO !nchunk > 1
+      DEALLOCATE(chunkWeight)
+    ENDSUBROUTINE makeContiguousDomains
+!
+!-------------------------------------------------------------------------------
+!> @brief get list of contiguous chunks in subdomain L
+!> @param thisGraph the graph object
+!> @param nv the number of vertices in subdomain L
+!> @param L the list of vertices in the subdomain
+!> @param nchunk the number of contiguous chunks
+!> @param ichunk the chunk index of each vertex in L
+!>
+    SUBROUTINE checkContiguity(thisGraph, nv, L, nchunk, ichunk)
```

Comment:
If this is only called by `makeContiguousDomains` perhaps state that in the comment description.